### PR TITLE
ispell: fix compilation on Darwin

### LINKED ISSUE #41948: ispell fails to build on Darwin
## Issue description

`ispell` fails to build on Darwin with the following error message:

```
make[1]: Leaving directory '/private/tmp/nix-build-ispell-3.3.02.drv-0/ispell-3.3.02'
chmod +x tryaffix
cc -O -c term.c
term.c:95:10: fatal error: 'termio.h' file not found
#include <termio.h>
         ^~~~~~~~~~
1 error generated.
term.c wouldn't compile.  Try inverting USG in local.h
make: *** [Makefile:588: term.o] Error 1
builder for '/nix/store/kv00qwnsbwi2zl6mvi9kyivkv0d4l8i9-ispell-3.3.02.drv' failed with exit code 2
error: build of '/nix/store/kv00qwnsbwi2zl6mvi9kyivkv0d4l8i9-ispell-3.3.02.drv' failed
```

### Steps to reproduce

```bash
nix-env -i -A nixpkgs.ispell
```

## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 17.6.0, macOS 10.13.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixpkgs-18.09pre133932.ee28e35ba37"`
 - channels(smaret): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

### PR DESCRIPTION
**Pull request by smaret**
###### Motivation for this change

Fixes #41948.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/text/ispell/default.nix [hunk 1]
```diff
@@ -22,8 +22,8 @@ stdenv.mkDerivation rec {
     ./patches/0030-Display-whole-multibyte-character.patch
   ];
   postPatch = ''
+    if [ ! stdenv.isDarwin ]; then echo "#define USG" >> local.h; fi
```
Comment (matthewbauer) [inline pkgs/tools/text/ispell/default.nix]: This should use optionalString

**Review by matthewbauer — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/ispell/default.nix (hunk 1 above)
Comment (LnL7) [inline pkgs/tools/text/ispell/default.nix]: that conditional isn't available in bash, you want something like this

```nix
{
  postPatch = ''
    cat >> local.h <<EOF
    ${stdenv.lib.optionalString (!stdenv.isDarwin) "#define USG"}
    #define TERMLIB "-lncurses"
    ...
    EOF
  '';
}
```

**Review by LnL7 — CHANGES_REQUESTED** [review]

**Comment by LnL7** [comment]
@GrahamcOfBorg build ispell

File: pkgs/tools/text/ispell/default.nix (hunk 1 above)
Comment (smaret) [reply pkgs/tools/text/ispell/default.nix]: Thanks for the suggestion.

**Review by LnL7 — APPROVED** [review]
Looks good now, doesn't even change the linux build.

**Pull request merged**